### PR TITLE
Update: TIPP Version 1.2.0 Released  

### PR DESCRIPTION
**Pull request by AnupamGautam**
Dear Admin,  

We have released new verison of TIPP 1.2.0. This new release includes the latest TIPPo method and additional flags for enhanced functionality.  

Please let me know if any further information is needed.  

Best regards,  
Anupam

Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by AnupamGautam** [comment]
@BiocondaBot please add label

File: recipes/tipp/meta.yaml [hunk 1]
```diff
@@ -39,16 +41,18 @@ requirements:
...
     - TIPP_plastid.v2.1.pl -h
     - TIPP_telomere.pl -h
+    - TIPPo.v2.3.pl -h
 
 about:
   home: https://github.com/Wenfei-Xian/TIPP
   license: "GPL-3.0-or-later"
   license_family: GPL3
-  summary: "TIPP: A User-Friendly Tool for De Novo Assembly of Organellar Genomes with HiFi Data"
+  summary: "TIPPo: A User-Friendly Tool for De Novo Assembly of Organellar Genomes with HiFi Data"
```
Comment (sjanssen2) [inline recipes/tipp/meta.yaml:55]: The semantics isn't totally clear to me. Is `TIPPo` a replacement of `TIPP` or should it be a different package? Your current git repo https://github.com/Wenfei-Xian/TIPP is names `TIPP` but the first header in the README.md says `TIPPo`?!

File: recipes/tipp/meta.yaml [hunk 2]
```diff
@@ -18,10 +18,12 @@ requirements:
     - {{ compiler('cxx') }}  # For compiling C++ code
     - make
     - kmc
+    - libzlib >=1.2.13,<1.3.0
```
Comment (sjanssen2) [inline recipes/tipp/meta.yaml:21]: I have not read through your commits. Is this rather strict pinning really necessary or could you relax at least the upper bound?

**Review by sjanssen2 — CHANGES_REQUESTED** [review]
Technically, your PR looks good to me. I just want to kindly ask you to clarify the tool name change.

File: recipes/tipp/meta.yaml (hunk 1 above)
Comment (AnupamGautam) [reply recipes/tipp/meta.yaml:55]: Dear @sjanssen2,

Thank you so much for your review. TIPP and TIPPo are essentially the same. When the manuscript was initially submitted, it was named TIPP_Plastid. However, during the review process, the reviewers suggested changes, and the name was updated to TIPPo.

The entire package, including TIPP, TIPPo, and TIPP_plastid, is part of the TIPP system. I specifically included TIPPo to help people easily cite or locate the manuscript.

Thank you again for your valuable feedback, and please let me know if further clarification is needed.

Best regards,
Anupam Gautam

File: recipes/tipp/meta.yaml (hunk 2 above)
Comment (AnupamGautam) [reply recipes/tipp/meta.yaml:21]: By default, the bioconda-build system was using `libzlib 1.3.1`, which caused issues with dependencies like `minimap2` and `graphaligner`, as these appear to have `libzlib` pinned to an older version. To resolve this, I explicitly pinned the `libzlib` version in the recipe.  

Previously, `libzlib` was not part of the `meta.yaml` file, and when I built it locally using `bioconda-build`, it still worked, as by default it is using`libzlib 1.2.13`. However, after pushing it to the current Bioconda build system, it started using the latest version of `libzlib` (1.3.1), which seems to have been updated very recently. This change triggered the errors, which are resolved by pinning the `libzlib` version.  I believe several other packages have also started encountering the same issue recently.

**Review by sjanssen2 — APPROVED** [review]
thanks @AnupamGautam for very elaborate answers. Makes total sense to me.

**Comment by AnupamGautam** [comment]
Thank you @sjanssen2

**Pull request merged**